File: src/main/java/github/zmilla93/core/utility/Platform.java
package github.zmilla93.core.utility;

/**
 * Enum representation of the 3 major OS platforms.
 * Platform.current returns the current OS.
 */
public enum Platform {

    UNKNOWN, WINDOWS, MAC, LINUX;

    public static final Platform current;
    public static final String currentOS;

    private final String name;


    static {
        currentOS = System.getProperty("os.name");
        final Platform debugPlatform = LINUX;
        if (debugPlatform != null) {
            current = debugPlatform;
        } else {
            if (currentOS.startsWith("Windows")) current = WINDOWS;
            else if (currentOS.startsWith("Mac")) current = MAC;
            else if (currentOS.startsWith("Linux")) current = LINUX;
            else current = UNKNOWN;
        }

    }

    Platform() {
        name = name().substring(0, 1).toUpperCase() + name().substring(1).toLowerCase();
    }

    @Override
    public String toString() {
        return name;
    }

}

File: src/main/java/github/zmilla93/core/utility/Platform.kt
package github.zmilla93.core.utility

import java.util.*

/**
 * Enum representation of the 3 major OS platforms.
 * Platform.current returns the current OS.
 */
enum class Platform {
    UNKNOWN, WINDOWS, MAC, LINUX;

    private val osName =
        name.substring(0, 1).uppercase(Locale.getDefault()) + name.substring(1).lowercase(Locale.getDefault())

    override fun toString(): String {
        return osName
    }

    companion object {
        @JvmField
        val current: Platform
        val currentOS: String = System.getProperty("os.name")
        val debugOS: Platform? = null

        init {
            if (debugOS != null) {
                current = debugOS
            } else {
                if (currentOS.startsWith("Windows")) current = WINDOWS
                else if (currentOS.startsWith("Mac")) current = MAC
                else if (currentOS.startsWith("Linux")) current = LINUX
                else current = UNKNOWN
            }
        }
    }
}
